pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -30,37 +30,33 @@ import (
 	binding "antrea.io/antrea/pkg/ovs/openflow"
 )
 
-// HandlePacketIn is the packetin handler registered to openflow by Antrea network
+// HandlePacketIn is the packetIn handler registered to openflow by Antrea network
 // policy agent controller. It performs the appropriate operations based on which
 // bits are set in the "custom reasons" field of the packet received from OVS.
 func (c *Controller) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
 	if pktIn == nil {
-		return errors.New("empty packetin for Antrea Policy")
+		return errors.New("empty packetIn for Antrea Policy")
 	}
 
-	matches := pktIn.GetMatches()
-	// Get custom reasons in this packet-in.
-	match := getMatchRegField(matches, openflow.CustomReasonField)
-	customReasons, err := getInfoInReg(match, openflow.CustomReasonField.GetRange().ToNXRange())
-	if err != nil {
-		return fmt.Errorf("received error while unloading customReason from reg: %v", err)
+	if len(pktIn.UserData) < 2 {
+		return errors.New("packetIn for Antrea Policy miss the required userdata")
 	}
-
-	// Use reasons to choose operations.
-	var checkCustomReason = func(customReasonMark *binding.RegMark) bool {
-		return customReasons&customReasonMark.GetValue() == customReasonMark.GetValue()
+	packetInOperations := pktIn.UserData[1]
+	// Choose operations.
+	var checkOperation = func(operation uint8) bool {
+		return packetInOperations&operation == operation
 	}
-	if checkCustomReason(openflow.CustomReasonLoggingRegMark) {
+	if checkOperation(openflow.PacketInNPLoggingOperation) {
 		if err := c.logPacket(pktIn); err != nil {
```

Comment:
You need to use `c.logPacketAction` here if that's the variable, instead of calling the method directly.